src/Http/Cookie/CookieInterface.php

```diff
@@ -79,24 +82,24 @@ public function getName(): string;
      *
      * @return array|string
      */
-    public function getValue();
+    public function getValue(): array|string;
 
     /**
      * Gets the cookie value as scalar.
      *
      * This will collapse any complex data in the cookie with json_encode()
      *
-     * @return mixed
+     * @return array|string
      */
-    public function getScalarValue();
+    public function getScalarValue(): array|string;
 
     /**
      * Create a cookie with an updated value.
      *
      * @param array|string $value Value of the cookie to set
      * @return static
      */
-    public function withValue($value);
+    public function withValue(array|string $value): static;
```

Comment:
Could add the `int|float|bool` here too. We could also narrow the type of value to only accept string|array, but I think accepting wider types and casting values to strings makes a more ergonomic API that is aligned with how CakePHP should work.